[llvm][Support] Add indirection to call correct validate(...) function

arromanoff opened the pull request:
Previously "yamlize" overload for validatedMappingTraits was
    unconditionally calling "MappingTraits<T>::validate" even if
    "MappingContextTraits<T, Context>" was passed to it. Therefore
    compilation failed when specifying
    "MappingContextTraits<T,Context>::validate()"

arrv-sc commented:
Small patch that makes it possible to use `validate(....)`  in `ContextMappingTraits`.

arrv-sc commented:
@RKSimon could you please take a look?

arrv-sc commented:
@aaupov

aaupov commented:
I'm sorry but I'm unfamiliar with the code.

arrv-sc commented:
@RKSimon, @topperc, take a look please

RKSimon reviewed (approved):
SGTM

arrv-sc commented:
@topperc

arrv-sc commented:
@topperc, please, review

arrv-sc commented:
I would highly appreciate you taking a look on this PR @topperc

topperc commented:
Is this a similar problem to what `detail::doMapping` is solving for the mapping function? Perhaps we could structure this the same way for consistency?

Commit:
[llvm][Support] Add indirection to call correct validate(...) function
    Previously "yamlize" overload for validatedMappingTraits was
    unconditionally calling "MappingTraits<T>::validate" even if
    "MappingContextTraits<T, Context>" was passed to it. Therefore
    compilation failed when specifying
    "MappingContextTraits<T,Context>::validate()"

arrv-sc commented:
> Is this a similar problem to what `detail::doMapping` is solving for the mapping function? Perhaps we could structure this the same way for consistency?

I actually agree with you. Updated.

topperc reviewed (approved):
LGTM

arrv-sc commented:
@RKSimon, merge?

RKSimon commented:
> @RKSimon, merge?

LGTM - do you mean you want me to merge it for you?

arrv-sc commented:
> LGTM - do you mean you want me to merge it for you?

Yes. Sorry if I'm not clear
